### PR TITLE
[OSDOCS-4426]: Egress IP support on RHOSP 

### PR DESCRIPTION
**Pull request by tmalove**
Version: 4.12+

Issue: [OSDOCS-3950](https://issues.redhat.com/browse/OSDOCS-3950)
Epic: [SDN-3027](https://issues.redhat.com/browse/SDN-3027)

Preview links: (updated 1/13): 
https://53440--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/configuring-egress-ips-ovn.html#nw-egress-ips-public-cloud-platform-considerations_configuring-egress-ips-ovn

https://53440--docspreview.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/assigning-egress-ips.html#nw-egress-ips-public-cloud-platform-considerations_egress-ips

QE review:
- [x ] QE has approved this change. (Huiran Wang)

### CODE REVIEW

**Comment by tmalove** [comment]
@andreaskaris  Will you review this update for 4.12? Thanks!

**Comment by tmalove** [comment]
Hi @andreaskaris, I updated the table. The note in the reverted commit is tracked in PR [53312](https://github.com/openshift/openshift-docs/pull/53312). I will look into the section that you suggested. Thanks!

**Comment by tmalove** [comment]
Hi @anuragthehatter will you review this change? Thanks!

**Comment by tmalove** [comment]
Hi @jechen0648 @huiran0826  Will one of you Ack this PR at your earliest? Anurag is on PTO through Tuesday. THanks!

**Comment by andreaskaris** [comment]
lgtm on my side to the change to the table   [Red Hat OpenStack Platform (RHOSP)     Yes ] which is the only change that I'm seeing in the doc

I'd still put this here under Public cloud platform considerations/OpenStack:
{code}
The {rh-openstack} egress IP address feature creates a neutron reservation port called `egressip-<IP address>`.
When an egress IP address on an OpenStack network is moved from one node to another, because of a node failover for example, the neutron reservation port is removed and recreated. As a result, a floating IP address might be bound to the port, creating a SNAT mapping of the egress IP address to a public IP address. Because SNAT breaks the floating IP address, a system administrator must recreate the floating IP address.
{code}

**Comment by tmalove** [comment]
Hi @andreaskaris this note is currently in these two places:
[OpenShift SDN default CNI network provider](http://file.rdu.redhat.com/tlove/sdn-neutron-4608-tlove/networking/openshift_sdn/assigning-egress-ips.html#nw-egress-ips-about_egress-ips)
[OVN-Kubernetes default CNI network provider](http://file.rdu.redhat.com/tlove/sdn-neutron-4608-tlove/networking/openshift_sdn/assigning-egress-ips.html#nw-egress-ips-about_egress-ips)

File: modules/nw-egress-ips-about.adoc [hunk 1]
```diff
@@ -20,6 +20,8 @@ To ensure that you can reliably allow access to the server from only that specif
 
 An egress IP address assigned to a namespace is different from an egress router, which is used to send traffic to specific destinations.
 
+Automatic attachment and detachment of egress IP addresses for traffic between nodes are available. This allows for traffic from many pods in namespaces to have a consistent source IP address to locations outside of the cluster. This also supports OpenShift SDN and OVN-Kubernetes as a default network provider.
+
```
Comment (ahardin-rh) [inline modules/nw-egress-ips-about.adoc:24]: OpenShift SDN and OVN-Kubernetes, which is the default networking plug-in in Red Hat OpenShift Networking in {product-title} {product-version}.

Approved language per Marc Curry in regard to rebrand. Thanks!

**Comment by tmalove** [comment]
Hi @andreaskaris Will you verify the note in the 'Public cloud platform considerations' sections? Thanks!

**Comment by andreaskaris** [comment]
> Hi @andreaskaris Will you verify the note in the 'Public cloud platform considerations' sections? Thanks!

The note lgtm, however I also found this here: https://issues.redhat.com/browse/OCPBUGS-4902  That when an admin user attaches the floating IP, instead of the reservation port being deleted, we might actually hang forever. So I'm not sure how  / if we should mention in 4.12 that we can actually do this ... perhaps we should just put it into a KCS and explain the different outcomes a bit further  ..

**Comment by tmalove** [comment]
> > Hi @andreaskaris Will you verify the note in the 'Public cloud platform considerations' sections? Thanks!
> 
> The note lgtm, however I also found this here: [issues.redhat.com/browse/OCPBUGS-4902](https://issues.redhat.com/browse/OCPBUGS-4902) That when an admin user attaches the floating IP, instead of the reservation port being deleted, we might actually hang forever. So I'm not sure how / if we should mention in 4.12 that we can actually do this ... perhaps we should just put it into a KCS and explain the different outcomes a bit further ..

@andreaskaris thanks for bringing up the Jira. We certainly do not want to document anything in our docs that is not 'solid'. I am discussing  this with my team and will get back to you soon!

**Comment by tmalove** [comment]
@andreaskaris will you verify that the table entry should be changed, however the paragraph should be removed? Thanks.

**Comment by andreaskaris** [comment]
I think that note is still needed but with all that we know now it should be:

~~~
The {rh-openstack} egress IP address feature creates a neutron reservation port called `egressip-<IP address>`. Using the same OpenStack user as the one used for the OpenShift cluster installation, you can assign a floating IP address to this reservation port to have a predictable SNAT address for egress traffic. When an egress IP address on an {rh-openstack} network is moved from one node to another, because of a node failover, for example, the neutron reservation port is removed and recreated. This means that the floating IP association is lost and you need to manually reassign the floating IP address to the new reservation port.
~~~

File: modules/nw-egress-ips-about.adoc [hunk 2]
```diff
@@ -80,6 +76,13 @@ The annotation value is an array with a single object with fields that provide t
 * `ifaddr`: Specifies the subnet mask for one or both IP address families.
 * `capacity`: Specifies the IP address capacity for the node. On AWS, the IP address capacity is provided per IP address family. On Azure and GCP, the IP address capacity includes both IPv4 and IPv6 addresses.
 
+Automatic attachment and detachment of egress IP addresses for traffic between nodes are available. This allows for traffic from many pods in namespaces to have a consistent source IP address to locations outside of the cluster. This also supports OpenShift SDN and OVN-Kubernetes, which is the default networking plug-in in Red Hat OpenShift Networking in {product-title} {product-version}.
```
Comment (mburke5678) [inline modules/nw-egress-ips-about.adoc:79]: s/ in Red Hat OpenShift Networking/for Red Hat OpenShift Networking ?? There are three instances of _in_ in close company. Reads a bit weird.

File: modules/nw-egress-ips-about.adoc [hunk 3]
```diff
@@ -80,6 +76,13 @@ The annotation value is an array with a single object with fields that provide t
...
+
+[NOTE]
+====
+The {rh-openstack} egress IP address feature creates a neutron reservation port called `egressip-<IP address>`. Using the same {rh-openstack} user as the one used for the {product-title} cluster installation, you can assign a floating IP address to this reservation port to have a predictable SNAT address for egress traffic. When an egress IP address on an {rh-openstack} network is moved from one node to another, because of a node failover, for example, the neutron reservation port is removed and recreated. This means that the floating IP association is lost and you need to manually reassign the floating IP address to the new reservation port.
```
Comment (mburke5678) [inline modules/nw-egress-ips-about.adoc:83]: Should _neutron_ be capitalized?

Comment (tmalove) [reply modules/nw-egress-ips-about.adoc]: Thanks @mburke5678 I made that capitalization change.

**Comment by mburke5678** [comment]
@tmalove PR needs a squash

**Comment by bergerhoffer** [comment]
@mburke5678 it looks like this PR wasn't CPd to 4.12. Issuing the cherry pick now and will ask for a merge exception there.

/cherrypick enterprise-4.12

**Pull request merged**